File: app/src/main/java/com/zack/enderplan/model/database/DatabaseOpenHelper.java
package com.zack.enderplan.model.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.zack.enderplan.util.Constant;

public class DatabaseOpenHelper extends SQLiteOpenHelper {

    private static final String CREATE_TABLE_PLAN = "create table " + Constant.PLAN + " (" +
            Constant.PLAN_CODE + " text primary key, " +
            Constant.CONTENT + " text, " +
            Constant.TYPE_CODE + " text, " +
            Constant.CREATION_TIME + " integer, " +
            Constant.DEADLINE + " integer, " +
            Constant.COMPLETION_TIME + " integer, " +
            Constant.STAR_STATUS + " integer, " +
            Constant.REMINDER_TIME + " integer)";

    private static final String CREATE_TABLE_TYPE = "create table " + Constant.TYPE + " (" +
            Constant.TYPE_CODE + " text primary key, " +
            Constant.TYPE_NAME + " text, " +
            Constant.TYPE_MARK_COLOR + " text, " +
            Constant.TYPE_MARK_PATTERN + " text, " +
            Constant.TYPE_SEQUENCE + " integer)";

    public DatabaseOpenHelper(Context context, String name, SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_TABLE_PLAN);
        db.execSQL(CREATE_TABLE_TYPE);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }
}

File: app/src/main/kotlin/me/imzack/app/end/model/database/DatabaseOpenHelper.kt
package me.imzack.app.end.model.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

import me.imzack.app.end.common.Constant

class DatabaseOpenHelper(context: Context, name: String, factory: SQLiteDatabase.CursorFactory?, version: Int) : SQLiteOpenHelper(context, name, factory, version) {

    private val CREATE_TABLE_PLAN = "create table " + Constant.PLAN + " (" +
            Constant.CODE + " text primary key, " +
            Constant.CONTENT + " text, " +
            Constant.TYPE_CODE + " text, " +
            Constant.CREATION_TIME + " integer, " +
            Constant.DEADLINE + " integer, " +
            Constant.COMPLETION_TIME + " integer, " +
            Constant.STAR_STATUS + " integer, " +
            Constant.REMINDER_TIME + " integer)"

    private val CREATE_TABLE_TYPE = "create table " + Constant.TYPE + " (" +
            Constant.CODE + " text primary key, " +
            Constant.NAME + " text, " +
            Constant.MARK_COLOR + " text, " +
            Constant.MARK_PATTERN + " text, " +
            Constant.SEQUENCE + " integer)"

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(CREATE_TABLE_PLAN)
        db.execSQL(CREATE_TABLE_TYPE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {

    }
}
